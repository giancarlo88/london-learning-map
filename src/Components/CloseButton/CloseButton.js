import React from 'react'

const CloseButton = ({ handleClick }) => (
  <button onClick={handleClick}>&times;</button>
)

export default CloseButton
