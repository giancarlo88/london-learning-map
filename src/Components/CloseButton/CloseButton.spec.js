import React from 'react'
import { shallow } from 'enzyme'

import CloseButton from './CloseButton'

describe(CloseButton.name, () => {
  beforeEach(() => {
    jest.clearAllMocks()
  })

  const defaultProps = {
    handleCloseClick: () => {}
  }

  describe('@renders', () => {
    it('should render', () => {
      const wrapper = shallow(<CloseButton {...defaultProps} />)
      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('@props', () => {
    describe('when clicked', () => {
      it('should fire the handleClick prop', () => {
        const handleClickMock = jest.fn()
        const props = { 
          ...defaultProps, 
          handleCloseClick: handleClickMock
        }
        const wrapper = shallow(<CloseButton {...props}/>)
        wrapper.simulate('click')
        expect(handleClickMock).toHaveBeenCalledTimes(1)
      })
    })
  })
})
