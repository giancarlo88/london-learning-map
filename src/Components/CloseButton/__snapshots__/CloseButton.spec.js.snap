// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CloseButton @renders should render 1`] = `
<button
  onClick={[Function]}
>
  Ã—
</button>
`;
